static/js/containers/pages/settings/AccountSettingsPage.js

```diff
@@ -30,35 +32,62 @@ type Props = {
     newPassword: string,
     confirmPassword: string
   ) => Promise<any>,
+  changeEmail: (newEmail: string, password: string) => Promise<any>,
   addUserNotification: Function,
   currentUser: User
 }
 
 export class AccountSettingsPage extends React.Component<Props> {
   async onSubmit(
-    { oldPassword, newPassword, confirmPassword }: ChangePasswordFormValues,
+    {
+      oldPassword,
+      newPassword,
+      confirmPassword,
+      email,
+      emailPassword
+    }: ChangePasswordFormValues,
     { setSubmitting, resetForm }: any
   ) {
-    const { addUserNotification, changePassword, history } = this.props
+    const {
+      addUserNotification,
+      changePassword,
+      changeEmail,
+      currentUser,
+      history
+    } = this.props
 
     try {
-      const response = await changePassword(
-        oldPassword,
-        newPassword,
-        confirmPassword
-      )
-
-      let alertText, color
-      if (response.status === 200) {
-        alertText = "Your password has been updated successfully."
-        color = "success"
+      let alertText, color, alertKey
+      if (currentUser && email !== currentUser.email) {
```

Comment:
I feel like this behavior of having email change and password change mutually exclusive, but having them share the same submit button in the UI, will be really confusing to your average end user.

Since there's only 1 button for both forms, I'd expect both my email and password to change if I filled out all the fields, but the actual behavior is:

- If I input a different email address than my current one, i get a change email confirmation sent, but the password change quietly fails
- If I input the same email address as my current one, my password is changed, but the email change quietly fails

Solutions that come to mind are to either have two submit buttons or perform both operations as expected and in either case nothing should quietly fail.

Reply:
Yes, it's confusing for user perspectives. We are doing whatever is provided in the UI.

By the way, I have added the feature of updating both email and password simultaneously.